### ISSUE TITLE
deskewing produced incorrect results when voxel size is different from 0.3

### ISSUE THREAD

**Issue by haesleinhuepf** [clEsperanto/pyclesperanto_prototype#172]
In pyclesperanto 0.15.4, there is an issue with deskewing. These two code snippetes should do pretty much the same:

```
deskewed_image = cle.deskew_y(pyramid_image,
                              voxel_size_x = 0.1449922,
                              voxel_size_y = 0.1449922,
                              voxel_size_z = 0.6,
                              angle_in_degrees = 30,
                             scale_factor=0.5)
```

```
scaled_pyramid_image = cle.scale(pyramid_image,factor_z=2,auto_size=False)

deskewed_scaled_pyramid_image = cle.deskew_y(scaled_pyramid_image,
                              voxel_size_x = 0.1449922,
                              voxel_size_y = 0.1449922,
                              voxel_size_z = 0.3,
                              angle_in_degrees = 30,
                                            scale_factor=0.5)
```

Note, in the second case, the voxel_size_z is half after the image has been scaled. One of the two versions appears sheared / skewed.

We can build in a scaling before deskewing as workaround, but obivously, it would be cool if the issue could be solved in the place where it happens, e.g. [here](https://github.com/clEsperanto/pyclesperanto_prototype/blob/master/pyclesperanto_prototype/_tier8/_AffineTransform3D.py#L310)

Thanks to @DrLachie for reporting!

**Comment by DrLachie** [comment]
I was in the process of putting this exact workaround into the llsz_napari plugin, but I guess it makes more sense to fix it here :) 

Thanks!

**Comment by haesleinhuepf** [comment]
I've tried to put the workaround scaling-step into the deskwing itself, because that would eliminate the need for an intermediate result `scaled_image`, but this didn't solve the issue. So we need to live with increased memory consumption, longer compute time and tiny interpolation artefacts for now :-/

**Comment by haesleinhuepf** [comment]
Ok, i've just deployed pyclesperanto_prototype==0.15.5 which contains this workaround. Would you mind testing @DrLachie ? 
Furthermore, we should keep this issue open until the actual underlying bug is fixed.

Thanks again for reporting the issue!

**Comment by DrLachie** [comment]
I can confirm it's working, but the noted memory overhead may be causing issues with some workflows

**Comment by haesleinhuepf** [comment]
Yes, I will come back to this asap. In two weeks likely. Thanks for testing!

**Comment by DrLachie** [comment]
Ok I've been tinkering. 
I propose redefining AffineTransform3D.deskew_y as: 

`    def _deskew_y(self, angle_in_degrees:float, voxel_size_x: float = 1,
                  voxel_size_y: float = 1, voxel_size_z: float = 1, scale_factor: float = 1):
        self.shear_in_x_plane(angle_y_in_degrees = 90 - angle_in_degrees)

        #defining shear factor wrt voxel size
        #This could potentiall replace shear_in_x_plane ? 
        shear_factor = math.sin((90-angle_in_degrees) * math.pi / 180.0) * (voxel_size_z / voxel_size_y)
        self._matrix[1,2] = shear_factor


        # rotate the stack to get proper Z-planes; rotate 90 - angle around X-axis
        #self.rotate(angle_in_degrees = 90 -angle_in_degrees, axis=0)
        
        # make voxels isotropic, calculate the new scaling factor for Z after shearing
        # https://github.com/tlambert03/napari-ndtiffs/blob/092acbd92bfdbf3ecb1eb9c7fc146411ad9e6aae/napari_ndtiffs/affine.py#L57
        new_dz = math.sin(angle_in_degrees * math.pi / 180.0) * voxel_size_z
        scale_factor_z = (new_dz / voxel_size_y) * scale_factor
        self.scale(scale_x=scale_factor, scale_y=scale_factor, scale_z=scale_factor_z)

        # correct orientation so that the new Z-plane goes proximal-distal from the objective.
        self.rotate(angle_in_degrees=90-angle_in_degrees, axis=0)

        return self`

This does swap the 0 and 1 axes of the resulting deskewed image though, wasn't sure if just chucking a np.swapaxes in was the way to go.  I'll be doing more testing tomorrow with a different dataset to see if I'm getting consistent results.

**Comment by haesleinhuepf** [comment]
This has been resolved for deskew_y in #174 

@DrLachie I just tried to adapt deskew_x in the same way, but didn't manage. Would you mind taking another look at this one? [This notebook](https://github.com/clEsperanto/pyclesperanto_prototype/blob/master/demo/transforms/deskew_x_test.ipynb) might be helpful. No hurry :-)

**Comment by haesleinhuepf** [comment]
Solved by @pr4deepr in 0.17.0

Thanks again! :-)
